onnxruntime/core/providers/dnnl/subgraph/dnnl_maxpoolgrad.h

```diff
@@ -144,82 +148,115 @@ class DnnlMaxPoolGrad : public DnnlKernel {
         dnnl::pooling_backward::desc(algo, *primitive_dst_md_, *diff_dst_md_,
                                      strides_mkl, kernel_mkl,
                                      padding_left_mkl, padding_right_mkl));
-
+    auto pool_fwd_pd = pool_fwd_->GetPrimitiveDesc();
     bwd_primitive_desc_ = onnxruntime::make_unique<dnnl::pooling_backward::primitive_desc>(
-        dnnl::pooling_backward::primitive_desc(*bwd_desc_, engine_to_use, *(pool_fwd_->GetPrimitiveDesc())));
-
-    if (mklnode_ptr_->parent_nodes.empty()) {
-      // Sub-graph's first node. Read input from input buffer
-      diff_dst_mem_ = onnxruntime::make_unique<dnnl::memory>(
-          dnnl::memory(bwd_primitive_desc_.get()->diff_dst_desc(), engine_to_use, nullptr));
-    } else {
-      // Sub-graph's inner node. set input to parent's output
-      diff_dst_mem_ = parents_[0].get()->primitive_dst_mem_;
-    }
+        dnnl::pooling_backward::primitive_desc(*bwd_desc_, engine_to_use, *pool_fwd_pd));
 
     primitive_src_desc_ = bwd_primitive_desc_.get()->diff_src_desc();
     primitive_dst_desc_ = bwd_primitive_desc_.get()->diff_dst_desc();
 
-    src_size_ = bwd_primitive_desc_.get()->diff_src_desc().get_size();
-    dst_size_ = bwd_primitive_desc_.get()->diff_dst_desc().get_size();
+    if (!gpu_available_) {
+      // reorder source memory for best performance (AVX512);
+      if (primitive_dst_desc_ != source_desc_) {
+        //dnnl::memory::dims src_dims(xgrad_shape_.GetDims().begin(), xgrad_shape_.GetDims().end());
+        auto pd = dnnl::memory::desc(source_desc_);
```

Comment:
Style nit: The following would be a little more compact... (I see this pattern in a few places).

```c++
dnnl::memory::desc pd(source_desc_);
```

Reply:
Thanks, fixed it in two places it showed up in this file.  I will have to keep my eye out for this pattern when doing code cleanup.